redisvl/schema/schema.py

```diff
@@ -181,17 +184,36 @@ def validate_and_create_fields(cls, values: Dict[str, Any]) -> Dict[str, Any]:
 
         input_fields = values.get("fields", [])
         prepared_fields: Dict[str, BaseField] = {}
-        # Handle old fields format temporarily
+
+        # Handle both list and dict formats for fields
         if isinstance(input_fields, dict):
-            raise ValueError("New schema format introduced; please update schema spec.")
-        # Process and create fields
-        for field_input in input_fields:
-            field = cls._make_field(index.storage_type, **field_input)
-            if field.name in prepared_fields:
-                raise ValueError(
-                    f"Duplicate field name: {field.name}. Field names must be unique across all fields."
-                )
-            prepared_fields[field.name] = field
+            # If fields is already a dict of BaseField instances, use it directly
+            for name, field in input_fields.items():
+                if isinstance(field, BaseField):
+                    prepared_fields[name] = field
+                elif isinstance(field, dict):
+                    # If it's a dict of field definitions, create fields
+                    field_obj = cls._make_field(index.storage_type, **field)
+                    if field_obj.name != name:
+                        raise ValueError(
+                            f"Field name mismatch: key '{name}' vs field name '{field_obj.name}'"
+                        )
+                    prepared_fields[name] = field_obj
+                else:
+                    raise ValueError(
+                        f"Invalid field type for '{name}': expected BaseField or dict"
+                    )
+        elif isinstance(input_fields, list):
```

Comment:
If you want to be less strict about the types, you could do `elif isinstance(input_fields, Sequence) and not isinstance(input_fields, (str, bytes))` (from collections.abc).